src/java.base/share/classes/java/lang/constant/MethodTypeDesc.java

```diff
@@ -204,11 +204,15 @@ static MethodTypeDesc of(ClassDesc returnDesc, ClassDesc... paramDescs) {
      * @return the human-readable descriptor for this method type
      */
     default String displayDescriptor() {
-        return String.format("(%s)%s",
-                             Stream.of(parameterArray())
-                                   .map(ClassDesc::displayName)
-                                   .collect(Collectors.joining(",")),
-                             returnType().displayName());
+        int count = parameterCount();
+        StringBuilder sb = new StringBuilder(24).append('(');
```

Comment:
24 is chosen by fair dice-roll? :)

Reply:
More or less: default capacity is 16, 24 is 1.5x that - a pretty typical growth factor. It also happens to be enough for very common descriptors such as `()Ljava/lang/Object;` while not too large to cause a regression on very small (and also very common) descriptors such as `()V`.